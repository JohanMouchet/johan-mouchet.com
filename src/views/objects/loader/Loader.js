// @flow

import * as React from "react";
import cx from "classnames";
import "./Loader.scss";

type Props = {
  size: "sm",
  metadata?: {
    contextClass?: string,
    additionalClasses?: Array<string>
  }
};

const Loader = (props: Props) => {
  const { size, metadata = {} } = props;

  const LoaderClass = cx(
    "o-loader",
    { [`o-loader--${size}`]: size },
    metadata.contextClass,
    metadata.additionalClasses
  );

  return <div className={LoaderClass}></div>;
};

export default Loader;
