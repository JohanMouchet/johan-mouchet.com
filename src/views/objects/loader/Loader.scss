@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";

.o-loader {
  $loader-speed: 1.5s;

  display: inline-block;
  height: t-vertical-rhythm(1.5, "rem");
  width: t-vertical-rhythm(1.5, "rem");
  outline: 2px solid $s-color-secondary;
  animation: loader-spin-animation $loader-speed infinite ease;

  @include t-wekkit-hdpi-small-dimension("outline-width", 2);

  @keyframes loader-spin-animation {
    50% {
      transform: rotate(45deg);
    }
    100% {
      transform: rotate(90deg);
    }
  }

  &::after {
    content: "";
    display: block;
    height: inherit;
    width: inherit;
    outline: inherit;
    animation: inherit;
    animation-name: loader-spin-animation-asynchronous;
  }

  @keyframes loader-spin-animation-asynchronous {
    50% {
      transform: rotate(90deg);
    }
    100% {
      transform: rotate(180deg);
    }
  }

  /** Small **/
  &--sm {
    height: t-vertical-rhythm(0.75, "rem");
    width: t-vertical-rhythm(0.75, "rem");
  }
}
