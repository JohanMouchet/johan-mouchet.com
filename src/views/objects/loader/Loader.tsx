import clsx, { ClassValue } from "clsx";
import styles from "./Loader.module.scss";

export const Loader = ({
  size,
  className,
  ...props
}: {
  size?: "sm";
  className?: ClassValue;
} & Omit<React.HTMLProps<HTMLDivElement>, "size">) => {
  const classNames = clsx(
    styles["o-loader"],
    size && styles[`o-loader--${size}`],
    className
  );

  return <div className={classNames} {...props} />;
};
