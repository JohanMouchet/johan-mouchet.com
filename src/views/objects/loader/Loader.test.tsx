import React from "react";
import { render } from "@testing-library/react";
import Loader from ".";

it("Renders <Loader /> without crashing", () => {
  expect(render(<Loader size="sm" />).container.firstChild).toMatchSnapshot();
});
