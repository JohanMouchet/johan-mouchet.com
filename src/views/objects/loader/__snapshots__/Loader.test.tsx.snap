// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Loader /> without crashing 1`] = `
<div
  className="o-loader o-loader--sm"
/>
`;
